compiler/ballerina-lang/src/main/java/org/ballerinalang/model/symbols/SymbolOrigin.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ballerinalang.model.symbols;
+
+/**
+ * Represents the various origination points of a symbol.
+ *
+ * @since 2.0.0
+ */
+public enum SymbolOrigin {
+
+    /**
+     * These are symbols which are defined internally by the compiler. e.g., symbols defined in the symbol table,
+     * symbols defined in lang lib
+     */
+    BUILTIN {
+        @Override
+        public SymbolOrigin toBIROrigin() {
+            return BUILTIN;
+        }
+
+        @Override
+        public byte value() {
+            return 1;
+        }
+    },
+
+    /**
+     * These are symbols for which there is a corresponding construct in the source file written by the user.
+     */
+    SOURCE {
+        @Override
+        public SymbolOrigin toBIROrigin() {
+            return COMPILED_SOURCE;
```

Comment:
Is this correct or should this be `SOURCE`?

Reply:
This is the intended value. The idea was to set a symbol's origin to `COMPILED_SOURCE` if it's coming from the BIR. So any `SOURCE` symbols become `COMPILED_SOURCE` symbols once the compilation is finished.

Reply:
Would a comment explaining this help?

Reply:
Ack, will add a comment